### ISSUE TITLE
Regional Attention Based Deep Feature for Image Retrieval

### ISSUE THREAD

**Issue by chullhwan-song** [chullhwan-song/Reading-Paper#40]
https://sglab.kaist.ac.kr/RegionalAttention/

**Comment by chullhwan-song** [comment]
### what ?
* R-MAC based deep feature 
* 여기에다가 attention 추가한 feature - context aware regional attention network 
* DIR연구와 달리, 단순히 pre-trained single-pass 개념의 학습 > softmax + imagenet으로 한듯.
   * DIR 나 delf는 neuralcode의 dataset 의 학습셋으로 이용
    * 여기서는, context aware regional attention network 학습하기 위해 imagenet을 적용한듯

### Regional Attentive Deep Feature
![image](https://user-images.githubusercontent.com/40360823/43821888-45911898-9b25-11e8-8dbc-e7dbfffa9fc4.png)
![image](https://user-images.githubusercontent.com/40360823/43822342-d2748f3c-9b26-11e8-8ff9-59e4032b2569.png)
![image](https://user-images.githubusercontent.com/40360823/43821940-7774f366-9b25-11e8-96fd-88ac8e5e6089.png)
   *  (figure.2) regional attention network+R-MAC 결합형태 수식
   * 빨간 부분이 R-MAC : ![image](https://user-images.githubusercontent.com/40360823/43822008-c9e98102-9b25-11e8-8aed-f1851a9d96b0.png)
   * 녹색 부분이 regional attention network : ![image](https://user-images.githubusercontent.com/40360823/43822024-d778e3e4-9b25-11e8-9c94-3675435973c7.png)
      * ![image](https://user-images.githubusercontent.com/40360823/43822043-e726bd34-9b25-11e8-8fd5-7b74e65d0468.png)
          *  ![image](https://user-images.githubusercontent.com/40360823/43822376-e40c9050-9b26-11e8-8b07-e87e6294dba7.png) 는 각 cropped feature map인 R에 대해, mean pooling 한 형태
          * ![image](https://user-images.githubusercontent.com/40360823/43822072-fadb4390-9b25-11e8-9de3-621367531eaa.png) 는 각각 linear transformation(fc)
          * ![image](https://user-images.githubusercontent.com/40360823/43822102-1183d95e-9b26-11e8-8566-cde99d9f4a1c.png) 에서 d는 crop 갯수(정확상, 논문에 나오지 않았음), k는 conv feature map에 대한 channel
          * 최종적으로  regional attention network는 d 차원의 vector인듯
* 여기서는 하나 개념을 추가 - figure 2에서 보면 **Context-aware regional vector** : ![image](https://user-images.githubusercontent.com/40360823/43822236-872dd95c-9b26-11e8-92b4-704ceb8b1aee.png)
![image](https://user-images.githubusercontent.com/40360823/43822214-70cf413c-9b26-11e8-85fd-da8ba7b74f07.png)
    * ![image](https://user-images.githubusercontent.com/40360823/43822222-790bf750-9b26-11e8-9821-061a31f49707.png) : concat 연산 (in the channel space)
    * regional attention network 의 input값인 ![image](https://user-images.githubusercontent.com/40360823/43822376-e40c9050-9b26-11e8-8b07-e87e6294dba7.png) 와 concat 연산을 통해 결합
    * ![image](https://user-images.githubusercontent.com/40360823/43822669-bef5042c-9b27-11e8-8320-9fdbd4050711.png) 은 convolution feature map 
    * convolution feature map과 ![image](https://user-images.githubusercontent.com/40360823/43822376-e40c9050-9b26-11e8-8b07-e87e6294dba7.png)(각 cropped feature map인 R에 대해, mean pooling 한 형태)와 concat한다는게...살짝 이해가 안감,     
    * 내가 제대로 이해했다면, convolution feature map는 WxHxC 이고 ![image](https://user-images.githubusercontent.com/40360823/43822376-e40c9050-9b26-11e8-8b07-e87e6294dba7.png)는 1x1xC 인데. dimension 구조가 다른데..어떻게 하지?? concat ?
 ### 실험
![image](https://user-images.githubusercontent.com/40360823/43822862-598cdab4-9b28-11e8-8f8b-a2f5be7933b7.png)
![image](https://user-images.githubusercontent.com/40360823/43822892-6eaf79e2-9b28-11e8-9fbb-d2ac5aa886b4.png)
